pkg/handler/mux.go

```diff
@@ -90,10 +95,15 @@ func (m *Mux) Invoke(ctx context.Context, payload []byte) (response []byte, err
 	}
 	ctx = logr.NewContext(ctx, logger)
 
+	span := trace.SpanFromContext(ctx)
```

Comment:
this seems more tracing related than handler/mux related? did this have to move here?

Reply:
Ah so the reason I moved it here is that the Otel wrapper directly calls this - if we want that span to show up as in error, and have the payload show up in the we need to get the context after the Otel wrapper span has modified the context (otherwise there will be no span associated with ctx, and the otel will create a span and will would add the payload and error handling to this new blank span). I therefore factored the code into the mux handler.